src/identities/mnemonic.ts

```diff
@@ -91,14 +101,14 @@ export default class Mnemonic extends Identity implements IdentityInterface {
    * increment the private member index +1.
    */
   private getNextKeypair(
-    isChange: boolean = false
+    isChange: boolean = false,
+    index: number
   ): { publicKey: Buffer; privateKey: Buffer } {
     const baseNode = this.masterPrivateKeyNode.derivePath(
       this.getCurrentDerivationPath(isChange)
     );
-    const wif: string = baseNode.deriveHardened(this.index).toWIF();
```

Comment:
at this point I think the name Next could be misleading. I would call deriveKeyWithIndex or something similar.